src/Parser/BalanceDruid/Modules/Spells/FullMoon.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { formatPercentage } from 'common/format';
+import Module from 'Parser/Core/Module';
+import SPELLS from 'common/SPELLS';
+
+class FullMoon extends Module {
+
+    fullMoonOrder = 0;
+    firstMoonTime = 0;
+    firstCastTime = 0;
+    firstCast = false;
+    firstMoonCast = false;
+    orderFound = false;
+
+    on_byPlayer_cast(event) {
+        if (!this.firstCast) {
+            this.firstCastTime = event.timestamp;
+            this.firstCast = true;
+        }
+
+        const spellId = event.ability.guid;
+        if (this.orderFound || (SPELLS.FULL_MOON.id !== spellId && SPELLS.HALF_MOON.id !== spellId && SPELLS.NEW_MOON.id !== spellId)) {
```

Comment:
Can we extract this logic out into a function, something like `isMoonSpell`?. Look at the Discipline `isAtonement` function for a similar premise.